test/integration/StreamrClient.test.js

```diff
@@ -62,6 +110,30 @@ describe('StreamrClient Connection', () => {
             await teardown()
         })
 
+        it('can reconnect after disconnect', (done) => {
+            client = createClient()
+            client.on('error', done)
+            client.connect()
+            client.once('connected', () => {
+                client.disconnect()
+            })
+            client.once('disconnected', () => {
+                client.connect()
+                client.once('connected', async () => {
+                    await client.disconnect()
+                    done()
+                })
+            })
+        })
+
+        it('can disconnect before connected', async (done) => {
+            client = createClient()
+            client.once('error', done)
+            client.connect()
+            await client.disconnect()
```

Comment:
Need to remove `error` listener before calling final `done`: `client.off('error', done)`. Jest is dumb and if `error` is fired later then the `done` that runs gets Jest very confused about what actually failed. We just resolved a week-long CI issue in `streamr-platform` this afternoon that stemmed from *exactly* this issue https://github.com/streamr-dev/streamr-platform/pull/329/files. 

Possible there's other error handlers in the `streamr-client` tests that should also be cleaned up.